[AUTOCUT] Gradle Check Flaky Test Report for RemoteStoreKafkaIT

## Flaky Test Report for `RemoteStoreKafkaIT`

Noticed the `RemoteStoreKafkaIT` has some flaky, failing tests that failed during **post-merge actions** and **timer trigerred run on main branch**.

### Details

| Git Reference | Merged Pull Request | Build Details | Test Name |
|---------------|----------------------|---------------|-----------|
| main | [null](https://github.com/opensearch-project/OpenSearch/pull/null) | [58898](https://build.ci.opensearch.org/job/gradle-check/58898/testReport/) <br><br>[60364](https://build.ci.opensearch.org/job/gradle-check/60364/testReport/) <br><br>[62476](https://build.ci.opensearch.org/job/gradle-check/62476/testReport/) <br><br>[62597](https://build.ci.opensearch.org/job/gradle-check/62597/testReport/) | `org.opensearch.plugin.kafka.RemoteStoreKafkaIT.testPauseAndResumeIngestion`<br><br>`org.opensearch.plugin.kafka.RemoteStoreKafkaIT.testErrorStrategy`<br><br>`org.opensearch.plugin.kafka.RemoteStoreKafkaIT.testRemoteSnapshotRestore` |
| 82ebf3b09a73f33a5f3d822e6b38bb138fe55e5e | [19074](https://github.com/opensearch-project/OpenSearch/pull/19074) | [62412](https://build.ci.opensearch.org/job/gradle-check/62412/testReport/) | `org.opensearch.plugin.kafka.RemoteStoreKafkaIT.testSegmentReplicationWithRemoteStore` |
| 2f416d3f2f2478daf479fcad3581b1eca2b8e67a | [19032](https://github.com/opensearch-project/OpenSearch/pull/19032) | [62279](https://build.ci.opensearch.org/job/gradle-check/62279/testReport/) | `org.opensearch.plugin.kafka.RemoteStoreKafkaIT.testPauseAndResumeIngestion` |
| 5fd0a36a8f3dff8affb56d2bd687c3c3b2262d49 | [18808](https://github.com/opensearch-project/OpenSearch/pull/18808) | [61327](https://build.ci.opensearch.org/job/gradle-check/61327/testReport/) | `org.opensearch.plugin.kafka.RemoteStoreKafkaIT.testPauseAndResumeIngestion` |
| a8b1734fa8ea43d8fb1d13cf49be54a3dc0df7e5 | [18087](https://github.com/opensearch-project/OpenSearch/pull/18087) | [57282](https://build.ci.opensearch.org/job/gradle-check/57282/testReport/) | `org.opensearch.plugin.kafka.RemoteStoreKafkaIT.testPauseAndResumeIngestion` |
| 4c76c1da1983cb8e04f0cf1769deac0ee57f7666 | [18027](https://github.com/opensearch-project/OpenSearch/pull/18027) | [57085](https://build.ci.opensearch.org/job/gradle-check/57085/testReport/) | `org.opensearch.plugin.kafka.RemoteStoreKafkaIT.testPauseAndResumeIngestion` |
| ddb435462333d244a30656318fa19cff42cdc294 | [17893](https://github.com/opensearch-project/OpenSearch/pull/17893) | [56370](https://build.ci.opensearch.org/job/gradle-check/56370/testReport/) | `org.opensearch.plugin.kafka.RemoteStoreKafkaIT.testErrorStrategy` |
| e8b975ddbd5cc7f0b39e50c95493a6c655a40933 | [17750](https://github.com/opensearch-project/OpenSearch/pull/17750) | [55870](https://build.ci.opensearch.org/job/gradle-check/55870/testReport/) | `org.opensearch.plugin.kafka.RemoteStoreKafkaIT.testSegmentReplicationWithRemoteStore` |
| 6b976ea0633a297e4c9b9cf7bf7263e6e1b658ad | [17836](https://github.com/opensearch-project/OpenSearch/pull/17836) | [55861](https://build.ci.opensearch.org/job/gradle-check/55861/testReport/) | `org.opensearch.plugin.kafka.RemoteStoreKafkaIT.testSegmentReplicationWithRemoteStore` |
| 17289b7685e72425e13051b1169518cdcafc4600 | [17663](https://github.com/opensearch-project/OpenSearch/pull/17663) | [54990](https://build.ci.opensearch.org/job/gradle-check/54990/testReport/) | `org.opensearch.plugin.kafka.RemoteStoreKafkaIT.testSegmentReplicationWithRemoteStore` |

The other pull requests, besides those involved in post-merge actions, that contain failing tests with the `RemoteStoreKafkaIT` class are:

- [17631](https://github.com/opensearch-project/OpenSearch/pull/17631)
- [18333](https://github.com/opensearch-project/OpenSearch/pull/18333)
- [18405](https://github.com/opensearch-project/OpenSearch/pull/18405)
- [18944](https://github.com/opensearch-project/OpenSearch/pull/18944)
- [18693](https://github.com/opensearch-project/OpenSearch/pull/18693)
- [17752](https://github.com/opensearch-project/OpenSearch/pull/17752)
- [17764](https://github.com/opensearch-project/OpenSearch/pull/17764)
- [17273](https://github.com/opensearch-project/OpenSearch/pull/17273)
- [17712](https://github.com/opensearch-project/OpenSearch/pull/17712)
- [17771](https://github.com/opensearch-project/OpenSearch/pull/17771)
- [17815](https://github.com/opensearch-project/OpenSearch/pull/17815)
- [17941](https://github.com/opensearch-project/OpenSearch/pull/17941)
- [19046](https://github.com/opensearch-project/OpenSearch/pull/19046)
- [17336](https://github.com/opensearch-project/OpenSearch/pull/17336)
- [17495](https://github.com/opensearch-project/OpenSearch/pull/17495)
- [17575](https://github.com/opensearch-project/OpenSearch/pull/17575)
- [17612](https://github.com/opensearch-project/OpenSearch/pull/17612)
- [17615](https://github.com/opensearch-project/OpenSearch/pull/17615)
- [17619](https://github.com/opensearch-project/OpenSearch/pull/17619)
- [17679](https://github.com/opensearch-project/OpenSearch/pull/17679)
- [17699](https://github.com/opensearch-project/OpenSearch/pull/17699)
- [17729](https://github.com/opensearch-project/OpenSearch/pull/17729)
- [17733](https://github.com/opensearch-project/OpenSearch/pull/17733)
- [17757](https://github.com/opensearch-project/OpenSearch/pull/17757)
- [17791](https://github.com/opensearch-project/OpenSearch/pull/17791)
- [17843](https://github.com/opensearch-project/OpenSearch/pull/17843)
- [17912](https://github.com/opensearch-project/OpenSearch/pull/17912)
- [18054](https://github.com/opensearch-project/OpenSearch/pull/18054)
- [18063](https://github.com/opensearch-project/OpenSearch/pull/18063)
- [18153](https://github.com/opensearch-project/OpenSearch/pull/18153)
- [18161](https://github.com/opensearch-project/OpenSearch/pull/18161)
- [18169](https://github.com/opensearch-project/OpenSearch/pull/18169)
- [18195](https://github.com/opensearch-project/OpenSearch/pull/18195)
- [18197](https://github.com/opensearch-project/OpenSearch/pull/18197)
- [18229](https://github.com/opensearch-project/OpenSearch/pull/18229)
- [18254](https://github.com/opensearch-project/OpenSearch/pull/18254)
- [18288](https://github.com/opensearch-project/OpenSearch/pull/18288)
- [18312](https://github.com/opensearch-project/OpenSearch/pull/18312)
- [18331](https://github.com/opensearch-project/OpenSearch/pull/18331)
- [18504](https://github.com/opensearch-project/OpenSearch/pull/18504)
- [18602](https://github.com/opensearch-project/OpenSearch/pull/18602)
- [18649](https://github.com/opensearch-project/OpenSearch/pull/18649)
- [18702](https://github.com/opensearch-project/OpenSearch/pull/18702)
- [18748](https://github.com/opensearch-project/OpenSearch/pull/18748)
- [18877](https://github.com/opensearch-project/OpenSearch/pull/18877)
- [18879](https://github.com/opensearch-project/OpenSearch/pull/18879)
- [18889](https://github.com/opensearch-project/OpenSearch/pull/18889)

For more details on the failed tests refer to [OpenSearch Gradle Check Metrics](https://metrics.opensearch.org/_dashboards/app/dashboards#/view/e5e64d40-ed31-11ee-be99-69d1dbc75083) dashboard.

@yupeng9 @varunbharadwaj Can one of you take a look here? These tests have been pretty flaky. Here's an example:

https://build.ci.opensearch.org/job/gradle-check/57507/consoleText

```
REPRODUCE WITH: ./gradlew ':plugins:ingestion-kafka:internalClusterTest' --tests "org.opensearch.plugin.kafka.RemoteStoreKafkaIT.testPauseAndResumeIngestion" -Dtests.seed=99C705639E2EC4B4 -Dtests.security.manager=false -Dtests.jvm.argline="-XX:TieredStopAtLevel=1 -XX:ReservedCodeCacheSize=64m" -Dtests.locale=bem-ZM -Dtests.timezone=SystemV/AST4ADT -Druntime.java=21

RemoteStoreKafkaIT > testPauseAndResumeIngestion FAILED
    java.lang.AssertionError: expected:<2> but was:<4>
        at __randomizedtesting.SeedInfo.seed([99C705639E2EC4B4:473224DAFADE6AA]:0)
        at org.junit.Assert.fail(Assert.java:89)
        at org.junit.Assert.failNotEquals(Assert.java:835)
        at org.junit.Assert.assertEquals(Assert.java:647)
        at org.junit.Assert.assertEquals(Assert.java:633)
        at org.opensearch.plugin.kafka.RemoteStoreKafkaIT.testPauseAndResumeIngestion(RemoteStoreKafkaIT.java:216)
```

--- a/plugins/ingestion-kafka/src/internalClusterTest/java/org/opensearch/plugin/kafka/RemoteStoreKafkaIT.java
+++ b/plugins/ingestion-kafka/src/internalClusterTest/java/org/opensearch/plugin/kafka/RemoteStoreKafkaIT.java
@@ -198,8 +198,9 @@ public class RemoteStoreKafkaIT extends KafkaIngestionBaseIT {
         assertTrue(pauseResponse.isShardsAcknowledged());
         waitForState(() -> {
             GetIngestionStateResponse ingestionState = getIngestionState(indexName);
-            return Arrays.stream(ingestionState.getShardStates())
-                .allMatch(state -> state.isPollerPaused() && state.pollerState().equalsIgnoreCase("paused"));
+            return ingestionState.getFailedShards() == 0
+                && Arrays.stream(ingestionState.getShardStates())
+                    .allMatch(state -> state.isPollerPaused() && state.pollerState().equalsIgnoreCase("paused"));
         });
 
         // verify ingestion state is persisted

--- a/plugins/ingestion-kafka/src/test/java/org/opensearch/plugin/kafka/KafkaUtils.java
+++ b/plugins/ingestion-kafka/src/test/java/org/opensearch/plugin/kafka/KafkaUtils.java
@@ -53,7 +53,7 @@ public class KafkaUtils {
         }
 
         // validates topic is created
-        await().atMost(10, TimeUnit.SECONDS).until(() -> checkTopicExistence(topicName, bootstrapServers));
+        await().atMost(60, TimeUnit.SECONDS).until(() -> checkTopicExistence(topicName, bootstrapServers));
     }
 
     public static boolean checkTopicExistence(String topicName, String bootstrapServers) {

